lib/elsa/group/manager/worker_supervisor.ex

```
@@ -45,8 +45,13 @@ defmodule Elsa.Group.Manager.WorkerSupervisor do
     connection = Keyword.fetch!(opts, :connection)
 
     children = [
-      {DynamicSupervisor,
-       [id: :worker_dynamic_supervisor, name: {:via, ElsaRegistry, {registry(connection), :worker_dynamic_supervisor}}]}
+      %{
+        id: :worker_dynamic_supervisor,
+        start:
+          {DynamicSupervisor, :start_link,
+           [[name: {:via, ElsaRegistry, {registry(connection), :worker_dynamic_supervisor}}]]},
+        restart: :transient
+      }
```

I had simply made a mistake here -- the tuple form of a child spec does not actually set the ID from the supervisor's perspective.  You have to use the struct form for that.

I believe the upshot of this mistake was that our calls to Supervisor.terminate_child below, were just failing silently.